lib/ansible/modules/windows/win_find.ps1

```diff
@@ -241,7 +241,7 @@ Function Get-FileStat($file) {
     $isdir = $false
     $isshared = $false
 
-    if ($attributes -contains 'ReparsePoint') {
+    if ($attributes -contains 'ReparsePoint' -and $attributes -contains 'Directory') {
```

Comment:
This probably isn't sustainable, a RepasePoint can be both a file or directory but the code (before the PR) incorrect assumes it is just a reparse point.

It sounds like this "deduped scenario" also uses them. Do you have a reproducer to create these types of files to test against?

Reply:
This is after I reproduced the issue

Reply:
What I'm trying to see is how you reproduced the referenced issue as the linked issue just shows the win_find task. Once we can reproduce it ourselves we can test and investigate further. ReparsePoints are more than just links but win_find right now incorrectly assumes they are.

Reply:
The issue is as such:
The win_find module has the assertion functions, one of them is the assert-filetype which checks if isdir is true of false basically.

dedupe in windows, as far as I recall when researching it in 2014 has a central block storage and original files are links of a sort to it.

specifically, here the deduped files have a reparse and sparse attributes and since the module assumed only dirs can be reparse points they marked `isdir = true` for all files/folders with reparse point attribute

Reply:
as per reproducing it, simply create a new volume on a windows server, enable dedup for it with 0 days till dedup and start an optimization job on the volume after moving some files to it

Reply:
> here the deduped files have a reparse and sparse attributes and since the module assumed only dirs can be reparse points they marked isdir = true for all files/folders with reparse point attribute

So we should be setting `$islnk = $true ` when the `ReparsePoint` attribute is set and only set `$isdir = $true` in the if statement below.

Reply:
I'll look at the meaning of islink in the module since most users and admins are not even aware that deduped files are links. we might break other options if we do that

Reply:
The meaning should correlate to `win_stat` but current the latter is a bit more advanced. The `islnk` should be set if the ReparsePoint is a symbolic link. This means the current logic in `win_find` that translate ReparsePoint to `islnk` is wrong but I don't think we should fix that in the current PR. It sounds like we now need a new return value like `isdedup` to report whether the file is a duplicate but right now we don't have a method to expose this function for you.

Once https://github.com/ansible/ansible/pull/56547 is merged in whatever form it takes we should change `win_find` to use that to get a better picture of the files it is querying. For now I would just fix up win_find to keep it as it is but do not set `isdir` when the attributes contains `ReparsePoint`.